rpmio/rpmpgp.c

```diff
@@ -1220,10 +1234,31 @@ int pgpPrtParamsSubkeys(const uint8_t *pkts, size_t pktlen,
 		pgpDigParamsFree(digps[count]);
 		continue;
 	    }
-	    count++;
+
+	    if (!params)
+		params = pgpDigParamsNew(0);
+
+	    if (decodePkt(p, (pend - p), &pkt) ||
+		pkt.tag != PGPTAG_SIGNATURE ||
+	        pgpPrtSig(0, pkt.body, pkt.blen, params) ||
+		params->sigtype != PGPSIGTYPE_SUBKEY_BINDING) {
+		pgpDigParamsFree(digps[count]);
```

Comment:
The condition is on the same indentation level as the actual code block, which has quite an obfuscating effect. Please fix that, either by indenting the condition deeper (as below), placing the opening brace on a separate line or other more creative means (helper variables, functions...)